apps/site/lib/site_web/controllers/trip_plan_controller.ex

```diff
@@ -96,6 +102,49 @@ defmodule SiteWeb.TripPlanController do
     %{leg | mode: mode_with_fares}
   end
 
+  @spec highest_highest_month_pass(Itinerary.t()) :: Fare.t() | nil
+  defp highest_highest_month_pass(%Itinerary{legs: legs}) do
+    legs
+    |> Enum.map(&highest_month_pass/1)
+    |> max_by_cents()
+  end
+
+  @spec highest_lowest_month_pass(Itinerary.t()) :: Fare.t() | nil
+  defp highest_lowest_month_pass(%Itinerary{legs: legs}) do
+    legs
+    |> Enum.map(&lowest_month_pass/1)
+    |> max_by_cents()
+  end
+
+  @spec highest_month_pass(Leg.t()) :: Fare.t() | nil
+  defp highest_month_pass(%Leg{mode: %PersonalDetail{}}), do: nil
+
+  defp highest_month_pass(%Leg{
+         mode: %TransitDetail{route_id: route_id, trip_id: trip_id},
+         from: %NamedPosition{stop_id: origin_id},
+         to: %NamedPosition{stop_id: destination_id}
+       }) do
+    Month.highest_pass(route_id, trip_id, origin_id, destination_id)
+  end
+
+  @spec lowest_month_pass(Leg.t()) :: Fare.t() | nil
+  defp lowest_month_pass(%Leg{mode: %PersonalDetail{}}), do: nil
+
+  defp lowest_month_pass(%Leg{
+         mode: %TransitDetail{route_id: route_id, trip_id: trip_id},
+         from: %NamedPosition{stop_id: origin_id},
+         to: %NamedPosition{stop_id: destination_id}
+       }) do
+    Month.lowest_pass(route_id, trip_id, origin_id, destination_id)
+  end
+
+  @spec max_by_cents([Fare.t() | nil]) :: Fare.t() | nil
+  defp max_by_cents(fares), do: Enum.max_by(fares, &cents_for_max/1, fn -> nil end)
+
+  @spec cents_for_max(Fare.t() | nil) :: non_neg_integer
+  defp cents_for_max(nil), do: 0
+  defp cents_for_max(%Fare{cents: cents}), do: cents
```

Comment:
The reason I made this just a little local private helper function is that I think it's only really useful in this context where `Enum.max` needs a value to compare, but you might have `nil` instead of a Fare struct. Generally, I don't think you'd want to say that an undefined fare is 0 cents.

We could also consider `max_by_cents` moving into `Fare`. It didn't seem worth it for this single usage, but if we ever need that functionality in another location I'd probably suggest that.